tests/snap_test.go

```diff
@@ -0,0 +1,201 @@
+	}
+
+	code := m.Run()
+	teardown()
+
+	os.Exit(code)
+}
+
+const sedCommand = `sed -i 's|/sys/devices/platform/axi/\*.pcie/\*.gpio/gpiochip4/dev|/sys/devices/platform/axi/*.pcie/*.gpio/gpiochip4/dev\n      - /sys/devices/platform/gpio-mockup.*/gpiochip*/dev|' squashfs-root/meta/snap.yaml`
+
+func authorizeGpioMock(path string) (string, error) {
+	_, stderr, err := utils.Exec(nil, "unsquashfs "+path)
+	if err != nil || stderr != "" {
+		log.Printf("stderr: %s", stderr)
+		log.Printf("err: %s", err)
+		return "", err
+	}
+
+	_, stderr, err = utils.Exec(nil, sedCommand)
+	if err != nil || stderr != "" {
+		log.Printf("stderr: %s", stderr)
+		log.Printf("err: %s", err)
+		return "", err
+	}
+
+	directory := filepath.Dir(path)
+	newPath := directory + "/mod_matter-pi-gpio-commander.snap"
+	_, stderr, err = utils.Exec(nil, "mksquashfs squashfs-root "+newPath)
+	if err != nil || stderr != "" {
+		log.Printf("stderr: %s", stderr)
+		log.Printf("err: %s", err)
+		return "", err
+	}
+
+	utils.Exec(nil, "rm -rf squashfs-root")
+	return newPath, nil
+}
+
+func setup() (teardown func(), err error) {
+	var newPath string
+
+	log.Println("[CLEAN]")
+	utils.SnapRemove(nil, snapName)
+
+	log.Println("[SETUP]")
+
+	teardown = func() {
+		log.Println("[TEARDOWN]")
+		utils.SnapDumpLogs(nil, start, snapName)
+
+		utils.Exec(nil, "rm "+newPath)
+
+		log.Println("Removing installed snap:", !utils.SkipTeardownRemoval)
+		if !utils.SkipTeardownRemoval {
+			utils.SnapRemove(nil, snapName)
+		}
+	}
+
+	// authorize gpio mock
+	if utils.LocalServiceSnap() {
+		newPath, err = authorizeGpioMock(utils.LocalServiceSnapPath)
+	}
+	if err != nil {
+		teardown()
+		return
+	}
+
+	// install matter-pi-gpio-commander
+	if utils.LocalServiceSnap() {
+		err = utils.SnapInstallFromFile(nil, newPath)
+	} else {
+		err = utils.SnapInstallFromStore(nil, snapName, utils.ServiceChannel)
+	}
+	if err != nil {
+		teardown()
+		return
+	}
+
+	if err = setupGPIO(); err != nil {
+		teardown()
+		return
+	}
+
+	// connect interfaces:
+	utils.SnapConnect(nil, snapName+":avahi-control", "")
+	utils.SnapConnect(nil, snapName+":bluez", "")
+	utils.SnapConnect(nil, snapName+":network", "")
+	utils.SnapConnect(nil, snapName+":network-bind", "")
+	utils.SnapConnect(nil, snapName+":custom-gpio", snapName+":custom-gpio-dev")
+
+	return
+}
+
+func getMockGPIO() (string, error) {
+	gpioChipNumber, stderr, err := utils.Exec(nil, "ls /dev/gpiochip* | sort -n | tail -n 1 | cut -d'p' -f3")
+	if err != nil || stderr != "" {
+		log.Printf("Failed to get mock gpio chip number, Error %s: %s", stderr, err)
+		return "", err
+	}
+	return gpioChipNumber, nil
+}
+
+func setupGPIO() error {
+
+	TestedGpioChip := os.Getenv(specificGpioChip)
+	TestedGpioLine := os.Getenv(specifGpioLine)
+
+	gpiochip := TestedGpioChip
+	var err error
+	if gpiochip == "" {
+		gpiochip, err = getMockGPIO()
+		if err != nil {
+			log.Printf("Failed to get mock gpio chip number: %s", err)
+			return err
```

Comment:
Errors messages should be returned to the caller to construct contextual and meaningful messages.
```suggestion
			return fmt.Errorf("failed to get mock gpio chip number: %s", err)
```